src/shared/varlink-io.systemd.Resolve.c

```diff
@@ -113,7 +113,7 @@ static SD_VARLINK_DEFINE_STRUCT_TYPE(
 static SD_VARLINK_DEFINE_STRUCT_TYPE(
                 ResolvedCanonical,
                 SD_VARLINK_DEFINE_FIELD(name, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
-                SD_VARLINK_DEFINE_FIELD(type, SD_VARLINK_STRING, 0),
+                SD_VARLINK_DEFINE_FIELD(type, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
```

Comment:
It can probably be nullable but I'm not sure why it gets resolved in the first place. `qotd.tcp` isn't valid: https://datatracker.ietf.org/doc/html/rfc6763#section-4.1.2

Reply:
In other words
```
{
  "name": null,
  "type": null,
  "domain": "beep.qotd.tcp.local"
}
```
is the last thing I'd expect in response to `beep._qotd._tcp.local`.

Reply:
I don't think it should be resolved in the first place but regardless of what's decided here the browser should be consistent with that too: https://github.com/systemd/systemd/pull/22532#issuecomment-2115192421. If ResolveService accepts that garbage the browser should be consistent with that too.